app/routes.go

```diff
@@ -8,16 +8,25 @@ func (app *App) InitRouter() {
 }
 
 func initialiseV1API(app *App) {
+
+	app.Router.NotFoundFunc(app.renderView(app.Handler404Error, false))
+
 	//REST API
-	app.Router.Get("/api/ping", app.handle(app.ping))
+	app.Router.Get("/api/ping", app.handle(app.ping, false))
 	// app.Router.Get("/api/todo/", app.handle(app.GetAllTodos))
 	//VIEW
-	app.Router.Get("/", app.renderView(app.RenderIndex))
-	app.Router.Get("/reviews/games", app.renderView(app.RenderReviewList))
-	app.Router.Get("/reviews/games/commentSubmit", app.renderView(app.SubmitComments))
-	app.Router.Get("/reviews/games/add", app.renderView(app.AddReviews))
-	app.Router.Post("/reviews/games/addSubmit", app.renderView(app.AddReviewsSubmit))
-	app.Router.Get("/reviews/games/:id", app.renderView(app.RenderReview))
+	app.Router.Get("/", app.renderView(app.RenderIndex, false))
+	app.Router.Get("/reviews/games", app.renderView(app.RenderReviewList, true))
+	app.Router.Get("/reviews/games/commentSubmit", app.renderView(app.SubmitComments, true))
+	app.Router.Get("/reviews/games/add", app.renderView(app.AddReviews, true))
+	app.Router.Post("/reviews/games/addSubmit", app.renderView(app.AddReviewsSubmit, true))
+	app.Router.Get("/reviews/games/:id", app.renderView(app.RenderReview, true))
```

Comment:
instead of sending in a `secureFlag` as part of your handler signature,you could do the following:
1.Define a method  `app.renderSecureView` like `app.handle` and let that function do everything for you.
2.`app.Router.Get("/reviews/games/:id", app.renderSecure(app.RenderReview)`
3.in your renderSecure you can do the check that you are doing in your handlers here
https://github.com/Arijeet-webonise/gameReview/pull/2/files#r169933445